docs/common/communication/passphrase-redesign-migration.md

```diff
@@ -1,94 +1,144 @@
-once the user entered theirs passphrase.
-
-### Supporting older firmwares
-
-With the new trezor-common code you should be able to support older firmwares as
-well. T1 should work out of the box!
-
-For TT, if the older firmware sends `PassphraseRequest.on_device` it is labeled
-as deprecated `_on_device` in the new Protobuf. However, that does not prevent
-you from using it:
-
-- In case the value is set to True, respond with an empty `PassphraseAck`.
-- In case the value is set to False, respond with `PassphraseAck` and set the
-passphrase in `passphrase` field.
-
-If you receive `PassphraseStateRequest`, now renamed to 
-`Deprecated_PassphraseStateRequest` just respond with
-`Deprecated_PassphraseStateAck`.
+The gist of the changes is:
 
-// TODO: describe the "state" in PassphraseState?
+- `PassphraseRequest.on_device` was deprecated, and renamed to `_on_device`. New Devices
+  will never send this field.
+- Corresponding field `PassphraseAck.on_device` was added.
+- `PassphraseStateRequest`/`PassphraseStateAck` messages were deprecated, and renamed
+  with a `Deprecated_` prefix. New Devices will not send or accept these messages.
+- `Initialize.state` was renamed to `Initialize.session_id`.
+- Corresponding field `Features.session_id` was added. New Devices will always send this
+  field in response to `Initialize` call.
+- A new value `Capability_PassphraseEntry` was added to the `Features.Capability`
+  enum. This capability will be sent from a Device that supports on-device passphrase
+  entry (currently only TT).
+
+### Restoring on-device entry for TT
+
+The Host software reacts to a `PassphraseRequest` message by prompting the user for a
+passphrase and sending it in the `PassphraseAck.passphrase` field.
+
+A new UI element should be added: when the passphrase prompt is displayed on Host, there
+should be an option to "enter passphrase on device". When the user selects this option,
+the Host must send a `PassphraseAck(passphrase=null, on_device=true)`.
+
+The "enter passphrase on device" option should be displayed when `Features.capabilities`
+contain the `Capability_PassphraseEntry` value, regardless of reported Trezor version of
+model. Firmwares older than 2.3.0 or 1.9.0 never set this value, so his ensures forwards
+and backwards compatibility.
+
+### Cross-version compatibility for TT
+
+TT version \< 2.3.0 will send `PassphraseRequest(_on_device=true)` if the user selected
+on-device entry. Neither T1 nor TT >= 2.3.0 will ever set this field to true.
+
+If the Host receives `PassphraseRequest(_on_device=true)`, it should immediately respond
+with `PassphraseAck()` with no fields set.
+
+TT version \< 2.3.0 will send `Deprecated_PassphraseStateRequest(state=[bytes])` after
+receiving `PassphraseAck`. The Host should immediately respond with
+`Deprecated_PassphraseStateAck()` with no fields set. If the Host does session
+management, it should store the value of `state` as the session ID.
```

Comment:
Note that the previous `state` had 64 bytes AFAIK and the new one has 32 bytes. I do not think it blocks anything, but it seems we should mention it.